randovania/games/samus_returns/generator/pool_creator.py

```diff
@@ -42,6 +43,19 @@ def pool_creator(results: PoolResults, configuration: BaseConfiguration, game: G
 
 
 def artifact_pool(game: GameDescription, config: MSRArtifactConfig) -> PoolResults:
+    # Check whether we have valid artifact requirements in configuration
+    max_artifacts = 0
+    if config.prefer_metroids:
+        max_artifacts = 25
+    elif config.prefer_stronger_metroids:
+        max_artifacts = 14
+    elif config.prefer_bosses:
+        max_artifacts = 3
+    elif not config.prefer_metroids and not config.prefer_stronger_metroids and not config.prefer_bosses:
+        max_artifacts = 39
+    elif config.required_artifacts > max_artifacts:
+        raise InvalidConfiguration("More Metroid DNA than allowed!")
```

Comment:
This is definetly wrong.
E.g. you set prefer metroids and stronger metroids but this is all `elif`, so  `max_artifacts` will be set to `25`.
BUT as it also uses else if in `elif config.required_artifacts > max_artifacts:`, the check is never done.
It should be `if config.required_artifacts > max_artifacts:` and that throws an `InvalidConfiguration` if you activate more than one option.
It needs to be pretty mucht the same logic as in `def num_preferred_locations`

Maybe a good reason to not leave this line uncovered from the tests :p